expression parser

fkleuver opened the pull request.

AureliaEffect commented:
This pull request **fixes 2 alerts** when merging cc9b50a0d7802fc728a69159f98a253fc2e2544b into 582c77c9561ce61619ef2644702b862671a8cbd7 - [view on LGTM.com](https://lgtm.com/projects/g/aurelia/aurelia/rev/pr-ec0cf028122ef16184a31ad58449636cff5357fe)

**fixed alerts:**

* 1 for Useless assignment to local variable
* 1 for Unused variable, import, function or class

---

*Comment posted by [LGTM.com](https://lgtm.com)*

AureliaEffect commented:
This pull request **fixes 2 alerts** when merging 9cae8f3a1eed3ef6a63c02684e9397282b1d53d1 into 582c77c9561ce61619ef2644702b862671a8cbd7 - [view on LGTM.com](https://lgtm.com/projects/g/aurelia/aurelia/rev/pr-0a084a39226439d97825325b65d261d3d1f197dd)

**fixed alerts:**

* 1 for Useless assignment to local variable
* 1 for Unused variable, import, function or class

---

*Comment posted by [LGTM.com](https://lgtm.com)*

AureliaEffect commented:
This pull request **fixes 4 alerts** when merging 57fa309f1dc265aefe20e1434ad2c6ee7071e149 into 582c77c9561ce61619ef2644702b862671a8cbd7 - [view on LGTM.com](https://lgtm.com/projects/g/aurelia/aurelia/rev/pr-b5c377b7d626728babb92fb31e4a48104cd273da)

**fixed alerts:**

* 2 for Comparison between inconvertible types
* 1 for Useless assignment to local variable
* 1 for Unused variable, import, function or class

---

*Comment posted by [LGTM.com](https://lgtm.com)*

AureliaEffect commented:
This pull request **fixes 4 alerts** when merging a4529709bcae85eb3397906b57c4124d4eaddd0d into 582c77c9561ce61619ef2644702b862671a8cbd7 - [view on LGTM.com](https://lgtm.com/projects/g/aurelia/aurelia/rev/pr-42b504da9d8f59fd7d90fdf0d7f91d9ae3ced643)

**fixed alerts:**

* 2 for Comparison between inconvertible types
* 1 for Useless assignment to local variable
* 1 for Unused variable, import, function or class

---

*Comment posted by [LGTM.com](https://lgtm.com)*

fkleuver commented:
@EisenbergEffect @bigopon 
Cutting off a new branch to continue the rest of the work on, this one can get merged if wanted. Summary of changes:
- Added an expression serializer to have a string format that contains all information about the AST (particularly useful for testing and debugging, probably not so much for portability due to the inefficiently large format)
- Changed all normal throws to throw Reporter error codes
- Significantly improved precision of testing all kinds of edge cases, combinations are much more thorough, several bugfixes as a result

AureliaEffect commented:
This pull request **fixes 4 alerts** when merging 7746447e8868b53d9010a36736c338a4214221d1 into 582c77c9561ce61619ef2644702b862671a8cbd7 - [view on LGTM.com](https://lgtm.com/projects/g/aurelia/aurelia/rev/pr-2caa1d5fb0516f49ef8a8054d30784153149ee27)

**fixed alerts:**

* 2 for Comparison between inconvertible types
* 1 for Useless assignment to local variable
* 1 for Unused variable, import, function or class

---

*Comment posted by [LGTM.com](https://lgtm.com)*

bigopon commented on packages/jit/src/expression-parser.ts:
```diff
@@ -49,9 +50,37 @@ export class ParserState {
+  MissingBindingBehavior = 113
+}
+
+const enum SemanticError {
```
Wow nice!
